.github/workflows/build-and-publish.yml

```diff
@@ -0,0 +1,42 @@
+name: Build and Publish to NPM
+on:
+    push:
+        branches: [master, development]
+
+permissions:
+    contents: read # for checkout
+
+jobs:
+    build_and_publish_npm:
+        name: Release
+        runs-on: ubuntu-latest
+        environment: Production
+        permissions:
+            contents: write # to be able to publish a GitHub release
+            issues: write # to be able to comment on released issues
+            pull-requests: write # to be able to comment on released pull requests
+            id-token: write # to enable use of OIDC for npm provenance
+        steps:
+            - name: Checkout
+              uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
+              with:
+                  fetch-depth: 0
+            - name: Setup Node.js
+              uses: actions/setup-node@b39b52d1213e96004bfcb1c61a8a6fa8ab84f3e8
+              with:
+                  node-version: '20'
+            - name: Update to latest npm (temporary fix for audit below)
+              run: npm install --global npm
+            - name: Install dependencies
+              run: npm clean-install
+            - name: Verify the integrity of provenance attestations and registry signatures for installed dependencies
```

Comment:
```suggestion
            - name: Verify the integrity of provenance attestations and registry signatures for installed dependencies
            if: success()
```

Reply:
I guess we need this 🤔

Reply:
By default every step has this condition, the next steps won't run if the previous step has failed

Reply:
Got it thanks 🙏